### DISCUSSION TITLE
Alternate toolbar style (FAB) for mobile

### DISCUSSION THREAD

**Discussion by chrisgurney** [chrisgurney/obsidian-note-toolbar#29] [Ideas]
From this [discussion thread](https://www.reddit.com/r/ObsidianMD/comments/1bzwk24/comment/kytfstk/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button) from [RIP_Pookie](https://www.reddit.com/user/RIP_Pookie/) on Reddit:

> If you're able to make a floating and expanding action button that is stickied by the bottom that would also be amazing.

It's got me thinking about this theme, although the FAB in this case just seems to toggle edit/reading mode: https://github.com/colineckert/obsidian-things

I would perhaps think of this as an optional appearance for mobile, based on the user's preference (and smaller screen sizes vs. tablet/iPad perhaps)... but I do like the idea as it makes the toolbar more thumb-accessible.

**Comment by MrComputron** [comment]
I would add that it would be extremely useful functionally if the FAB was available at root so it was always available.

**Reply by chrisgurney** [reply]
@MrComputron Thank you!

FYI Folder Mappings now supports, as of the v1.5 beta:
- `/` = map toolbars for notes that just live in the root folder
- `*` = apply toolbars for all of your notes (you can still override this with more specific folder mappings or the property)

**Reply by MrComputron** [reply]
Phenomenal...I'll check it out

**Comment by chrisgurney** [comment]
@MrComputron (and anybody else reading this): 

I'm curious, what do you use this button for on mobile (circled in red)? Do you have it mapped to anything in particular? And do you use it often?

![CleanShot 2024-04-24 at 09 55 42@2x](https://github.com/chrisgurney/obsidian-note-toolbar/assets/17337703/1bd3ade9-122c-46e2-aaba-0759dbe7edd9)

**Reply by MrComputron** [reply]
I almost never use it. I actually normally forget it's even there.

**Reply by MrComputron** [reply]
It would be very useful if you could customize the actions and buttons on there but I can't figure out even what this toolbar is called...I think it's one of the reasons I installed Commander as I wanted to customize it (which failed). It did allow me to put my favorite actions in the top bar instead.

**Reply by chrisgurney** [reply]
Thanks! 

It's actually a shortcut that, by default, opens a menu with the options that you have in your Ribbon Menu (the left sidebar on desktop).

Customize it via:
_Appearance > Ribbon Menu > Manage_

I found out that I can use it to launch a menu that I can put the toolbar's items in. Version 1 of a better mobile experience!

**Comment by chrisgurney** [comment]
Check out the beta for floating buttons and more for mobile!

[1.7 Announcement](https://github.com/chrisgurney/obsidian-note-toolbar/discussions/62)

**Comment by chrisgurney** [comment]
@MrComputron The ability to toggle visibility of icons and labels by platform, floating buttons, and more for mobile is now available in the [1.7 release](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.7.21), available in Community Plugins. Thank you for your feedback on this feature request!

**Reply by MrComputron** [reply]
I just tested it out and it's great!
I had to reconfigure the order so that the Note Toolbar button is positioned for immediate selection upon initial button press without scrolling.
I assume that there is no way to simply add the button to the bottom toolbar because that would remove a button press for NT.

![Screenshot_20240511_112117_Obsidian](https://github.com/chrisgurney/obsidian-note-toolbar/assets/46870771/ebd74b27-9f6e-477d-bb4c-9911f3b54d50)

I could remove all other buttons from toolbar to make it even more dedicated to the note Toolbar as I don't use any of those in any case, but it would still be an extra step.

Regardless, I will continue to test it out...this implementation looks very promising.

**Reply by chrisgurney** [reply]
Unfortunately there's no way to add an additional button to the nav bar that I'm aware of, but you _can_ replace the rightmost icon with Note Toolbar if you wish (set Note Toolbar as the "Quick access ribbon item").

**Reply by chrisgurney** [reply]
> I had to reconfigure the order so that the Note Toolbar button is positioned for immediate selection upon initial button press without scrolling.

I didn't quite understand what you meant by this. Can you explain?

**Reply by MrComputron** [reply]
I reorganized the vertical stack of sub items so that Note Toolbar always appeared at the very bottom (without scrolling down further). It doesn't matter though because setting Note Toolbar as quick access ribbon item is **exactly** what I was looking for!
It might help to put a bit of a how-to into the plug-in documentation for this setup because I found it difficult to find the settings for enabling it and it's a beautiful implementation of Note Toolbar for longer notes.

**Reply by chrisgurney** [reply]
That's great feedback, thank you! 

It's in the user guide, but I'll see if I can fit it in the plugin settings UI in a natural place, as well.
